src/core/main.c

```diff
@@ -1294,6 +1294,28 @@ static int bump_unix_max_dgram_qlen(void) {
         return 1;
 }
 
+static int fixup_environment(void) {
+        _cleanup_free_ char *term = NULL;
+
+        /* When started as PID1, the kernel uses /dev/console
+         * for our stdios and uses TERM=linux whatever the
+         * backend device used by the console. We try to make
+         * a better guess here since some consoles might not
+         * have support for color mode for example.
+         *
+         * However if TERM was configured through the kernel
+         * command line then leave it alone. */
+
+        if (get_proc_cmdline_key("TERM=", &term) == 0)
+                term = strdup(default_term_for_tty("/dev/console") + 5);
+
+        if (!term)
+                return -1;
```

Comment:
please do not make up error codes such as "-1", see CODING_STYLE. Please make thus function work like our usual functions, and return negative errno values on error. In fact please return the actual error returned by get_proc_cmdline_key() if it returns negative. And return -ENOMEM if strdup() fails...